adapt to changes in unliftio-core-0.2.0.1

In particular

https://github.com/fpco/unliftio/commit/d327ae40853635363c3eae8196bb1ee9e88b5c57

Which results in 

```
Haskell/Ide/Engine/PluginsIdeMonads.hs:588:16: error:
    The INLINE pragma for ‘askUnliftIO’ lacks an accompanying binding
      (The INLINE pragma must be given where ‘askUnliftIO’ is declared)
    |
588 |     {-# INLINE askUnliftIO #-}
    |                ^^^^^^^^^^^

Haskell/Ide/Engine/PluginsIdeMonads.hs:589:5: error:
    ‘askUnliftIO’ is not a (visible) method of class ‘MonadUnliftIO’
    |
589 |     askUnliftIO = Ghc $ \s ->
    |     ^^^^^^^^^^^

Haskell/Ide/Engine/PluginsIdeMonads.hs:600:16: error:
    The INLINE pragma for ‘askUnliftIO’ lacks an accompanying binding
      (The INLINE pragma must be given where ‘askUnliftIO’ is declared)
    |
600 |     {-# INLINE askUnliftIO #-}
    |                ^^^^^^^^^^^

Haskell/Ide/Engine/PluginsIdeMonads.hs:601:5: error:
    ‘askUnliftIO’ is not a (visible) method of class ‘MonadUnliftIO’
    |
601 |     askUnliftIO = GhcT $ \s ->
    |     ^^^^^^^^^^^
```

--- a/cabal.project
+++ b/cabal.project
@@ -16,8 +16,6 @@ package haskell-ide-engine
 
 -- Match the flag settings we use in stac builds
 constraints:
-        -- to avoid temporary a breaking change: https://github.com/haskell/haskell-ide-engine/issues/1680
-        unliftio-core < 0.2.0.0,
         haskell-ide-engine +pedantic,
         hie-plugin-api     +pedantic
 

--- a/cabal.project-8.8.2
+++ b/cabal.project-8.8.2
@@ -24,8 +24,6 @@ package haskell-ide-engine
 
 -- Match the flag settings we use in stac builds
 constraints:
-        -- to avoid temporary a breaking change: https://github.com/haskell/haskell-ide-engine/issues/1680
-        unliftio-core < 0.2.0.0,
         haskell-ide-engine +pedantic,
         hie-plugin-api     +pedantic
 

--- a/hie-plugin-api/Haskell/Ide/Engine/PluginsIdeMonads.hs
+++ b/hie-plugin-api/Haskell/Ide/Engine/PluginsIdeMonads.hs
@@ -585,11 +585,6 @@ instance MonadTrans GhcT where
 
 
 instance MonadUnliftIO Ghc where
-    {-# INLINE askUnliftIO #-}
-    askUnliftIO = Ghc $ \s ->
-                  withUnliftIO $ \u ->
-                  return (UnliftIO (unliftIO u . flip unGhc s))
-
     {-# INLINE withRunInIO #-}
     withRunInIO inner =
       Ghc $ \s ->
@@ -597,11 +592,6 @@ instance MonadUnliftIO Ghc where
       inner (run . flip unGhc s)
 
 instance MonadUnliftIO (GhcT IdeM) where
-    {-# INLINE askUnliftIO #-}
-    askUnliftIO = GhcT $ \s ->
-                  withUnliftIO $ \u ->
-                  return (UnliftIO (unliftIO u . flip unGhcT s))
-
     {-# INLINE withRunInIO #-}
     withRunInIO inner =
       GhcT $ \s ->

--- a/hie-plugin-api/hie-plugin-api.cabal
+++ b/hie-plugin-api/hie-plugin-api.cabal
@@ -57,6 +57,7 @@ library
                      , haskell-lsp == 0.20.*
                      , hslogger
                      , unliftio
+                     , unliftio-core ^>= 0.2.0.1
                      , monad-control
                      , mtl
                      , process

--- a/stack-8.4.2.yaml
+++ b/stack-8.4.2.yaml
@@ -49,6 +49,7 @@ extra-deps:
 - parser-combinators-1.2.1
 - profunctors-5.5.1
 - pretty-show-1.8.2
+- resourcet-1.2.3 # forced by unliftio-core >= 0.2
 - rope-utf16-splay-0.3.1.0
 - simple-sendfile-0.2.30 # for network and network-bsd
 - socks-0.6.1 # for network and network-bsd

--- a/stack-8.4.2.yaml
+++ b/stack-8.4.2.yaml
@@ -55,7 +55,8 @@ extra-deps:
 - syz-0.2.0.0
 - type-equality-1
 - unix-compat-0.5.2
-- unliftio-0.2.12
+- unliftio-0.2.12.1
+- unliftio-core-0.2.0.1
 - unordered-containers-0.2.10.0
 - yaml-0.11.2.0
 - th-abstraction-0.3.1.0

--- a/stack-8.4.3.yaml
+++ b/stack-8.4.3.yaml
@@ -74,7 +74,8 @@ extra-deps:
 - topograph-1
 - type-equality-1
 - unix-compat-0.5.2
-- unliftio-0.2.12
+- unliftio-0.2.12.1
+- unliftio-core-0.2.0.1
 - unordered-containers-0.2.10.0
 - vector-0.12.1.2
 - yaml-0.11.2.0

--- a/stack-8.4.3.yaml
+++ b/stack-8.4.3.yaml
@@ -61,6 +61,7 @@ extra-deps:
 - parser-combinators-1.2.1
 - profunctors-5.5.1
 - quickcheck-instances-0.3.22
+- resourcet-1.2.3 # forced by unliftio-core >= 0.2
 - rope-utf16-splay-0.3.1.0
 - semialign-1.1
 - semigroupoids-5.3.4

--- a/stack-8.4.4.yaml
+++ b/stack-8.4.4.yaml
@@ -49,6 +49,7 @@ extra-deps:
 - parser-combinators-1.2.1
 - pretty-show-1.9.5
 - profunctors-5.5.1
+- resourcet-1.2.3 # forced by unliftio-core >= 0.2
 - rope-utf16-splay-0.3.1.0
 - simple-sendfile-0.2.30 # for network and network-bsd
 - socks-0.6.1 # for network and network-bsd

--- a/stack-8.4.4.yaml
+++ b/stack-8.4.4.yaml
@@ -54,7 +54,8 @@ extra-deps:
 - socks-0.6.1 # for network and network-bsd
 - syz-0.2.0.0
 - unix-compat-0.5.2
-- unliftio-0.2.12
+- unliftio-0.2.12.1
+- unliftio-core-0.2.0.1
 - unordered-containers-0.2.10.0
 - yaml-0.11.2.0
 - th-abstraction-0.3.1.0

--- a/stack-8.6.4.yaml
+++ b/stack-8.6.4.yaml
@@ -35,6 +35,7 @@ extra-deps:
 - multistate-0.8.0.1
 - ormolu-0.0.3.1
 - parser-combinators-1.2.1
+- resourcet-1.2.3 # forced by unliftio-core >= 0.2
 - rope-utf16-splay-0.3.1.0
 - syz-0.2.0.0
 - temporary-1.2.1.1

--- a/stack-8.6.4.yaml
+++ b/stack-8.6.4.yaml
@@ -40,7 +40,8 @@ extra-deps:
 - temporary-1.2.1.1
 - time-compat-1.9.2.2
 - unix-compat-0.5.2
-- unliftio-0.2.12
+- unliftio-0.2.12.1
+- unliftio-core-0.2.0.1
 - unordered-containers-0.2.10.0
 - yaml-0.11.2.0
 # To make build work in windows 7

--- a/stack-8.6.5.yaml
+++ b/stack-8.6.5.yaml
@@ -39,6 +39,7 @@ extra-deps:
 - optparse-applicative-0.15.1.0
 - ormolu-0.0.3.1
 - parser-combinators-1.2.1
+- resourcet-1.2.3 # forced by unliftio-core >= 0.2
 - semialign-1.1
 - temporary-1.2.1.1
 - topograph-1

--- a/stack-8.6.5.yaml
+++ b/stack-8.6.5.yaml
@@ -42,6 +42,8 @@ extra-deps:
 - semialign-1.1
 - temporary-1.2.1.1
 - topograph-1
+- unliftio-0.2.12.1
+- unliftio-core-0.2.0.1
 
 flags:
   haskell-ide-engine:

--- a/stack-8.8.1.yaml
+++ b/stack-8.8.1.yaml
@@ -30,6 +30,7 @@ extra-deps:
 - ilist-0.3.1.0
 - monad-dijkstra-0.1.1.2
 - ormolu-0.0.3.1
+- resourcet-1.2.3 # forced by unliftio-core >= 0.2
 - semigroups-0.18.5
 - temporary-1.2.1.1
 - unliftio-0.2.12.1

--- a/stack-8.8.1.yaml
+++ b/stack-8.8.1.yaml
@@ -32,6 +32,8 @@ extra-deps:
 - ormolu-0.0.3.1
 - semigroups-0.18.5
 - temporary-1.2.1.1
+- unliftio-0.2.12.1
+- unliftio-core-0.2.0.1
 
 flags:
   haskell-ide-engine:

--- a/stack-8.8.2.yaml
+++ b/stack-8.8.2.yaml
@@ -35,6 +35,7 @@ extra-deps:
 - ilist-0.3.1.0
 - monad-dijkstra-0.1.1.2
 - ormolu-0.0.3.1
+- resourcet-1.2.3 # forced by unliftio-core >= 0.2
 - semigroups-0.18.5
 - temporary-1.2.1.1
 - unliftio-0.2.12.1

--- a/stack-8.8.2.yaml
+++ b/stack-8.8.2.yaml
@@ -37,6 +37,8 @@ extra-deps:
 - ormolu-0.0.3.1
 - semigroups-0.18.5
 - temporary-1.2.1.1
+- unliftio-0.2.12.1
+- unliftio-core-0.2.0.1
 
 flags:
   haskell-ide-engine:

--- a/stack.yaml
+++ b/stack.yaml
@@ -35,6 +35,8 @@ extra-deps:
 - ormolu-0.0.3.1
 - semigroups-0.18.5
 - temporary-1.2.1.1
+- unliftio-0.2.12.1
+- unliftio-core-0.2.0.1
 
 flags:
   haskell-ide-engine:

--- a/stack.yaml
+++ b/stack.yaml
@@ -33,6 +33,7 @@ extra-deps:
 - ilist-0.3.1.0
 - monad-dijkstra-0.1.1.2
 - ormolu-0.0.3.1
+- resourcet-1.2.3 # forced by unliftio-core >= 0.2
 - semigroups-0.18.5
 - temporary-1.2.1.1
 - unliftio-0.2.12.1

